bindings/nodejs/lib/types/block/core/block_wrapper.ts

```diff
@@ -0,0 +1,129 @@
+// Copyright 2023 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+import { IssuerId, SlotCommitmentId } from '../id';
+import { Ed25519Signature } from '../signature';
+
+/**
+ * All of the block types.
+ */
+enum BlockType {
+    /// A Basic block.
+    Basic = 0,
+    /// A Verification block.
+    Verification = 1,
+}
+
+abstract class BlockWrapper {
+    /**
+     * The type of block
+     */
+    readonly type: BlockType;
+
+    /**
+     * Protocol version of the block.
+     */
+    readonly protocolVersion: number;
+    /**
+     * Network identifier.
+     */
+    readonly networkId: bigint;
+    /**
+     * The time at which the block was issued. It is a Unix timestamp in nanoseconds.
+     */
+    readonly issuingTime: bigint;
+    /**
+     * The identifier of the slot to which this block commits.
+     */
+    readonly slotCommitmentId: SlotCommitmentId;
+    /**
+     * The slot index of the latest finalized slot.
+     */
+    readonly latestFinalizedSlot: bigint;
```

Comment:
this should be of type `SlotIndex`, also after https://github.com/iotaledger/iota-sdk/pull/1005 is merged.